### PR TITLE
Feature/add association

### PR DESCRIPTION
**Pull request by thasipablo**
### Project features

- Created a class Classroom
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`.
- Created a class `Book`
- Created a class `Rental`
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: lib/book.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
```
Comment (frank1738) [inline lib/book.rb]: - [ ] Kindly add a method of adding rentals to your `book` class to achieve `many-to-many-relationship`
- [ ] Also do the same for your `Person` class

### Example 

```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
 end
```

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @thasipablo , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Organized work
✅ Most of the associations are set correctly 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by thasipablo** [comment]
Thank you @frank1738 . I have added the methods

File: lib/person.rb [hunk 2]
```diff
@@ -27,4 +28,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(person, date)
+    @rentals << Rental.new(date, self, person)
+  end
```
Comment (Owusu-Desmond) [inline lib/person.rb:34]: - [ ] Please ensure that this `add_rental` method takes in the book and the date as parameters then the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association :pray:

File: lib/book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    @rentals << Rental.new(date, self, person)
+  end
```
Comment (Owusu-Desmond) [inline lib/book.rb:13]: - [ ] Kindly ensure that the `add_rental` method takes in the person and the date  as parameters then the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.` :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @thasipablo , :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by thasipablo** [comment]
Thank you @Owusu-Desmond. The add_rental was updated

**Review by Munsa1 — APPROVED** [review]
Hi @thasipablo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**